src/network.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{Message, PeerId, TransactionSet};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(u8)]
+pub enum Consensus {
+    Inconclusive = 0x00,
+    Consensus = 0x01,
+    Reject = 0x02,
+}
+
+impl Consensus {
+    pub const fn from_u8(byte: u8) -> Self {
+        match byte {
+            0x00 => Self::Inconclusive,
+            0x01 => Self::Consensus,
+            _ => Self::Reject,
+        }
+    }
+}
+
+pub trait Network {
+    type PeerId: PeerId;
+    type Message: Message<PeerId = Self::PeerId>;
+
+    fn increment_height(height: u64) -> u64;
+    fn is_council(&self, height: u64, peer: &Self::PeerId) -> bool;
+    fn peers(&self, height: u64) -> usize;
+    fn proposer(&self, height: u64) -> &Self::PeerId;
+    fn transaction_set(&self) -> TransactionSet;
```

Comment:
I'm wondering if we should even include this. Since the core of pbft operates at the block-level, interactions with the txpool could be handled at the implementation layer of the `Node` trait.

Reply:
We want to avoid serialization/deserialization of types based on agnostic bytes. Using the transaction here is a way to bring this spec closer to our protocol, but it indeed can be replaced by associated types of the trait that satisfies some criteria

Reply:
Since the block type is already generic, and methods like `new_block` and `validate_block` are also generic, I don't see the need for the pbft algorithm to manually handle specific txs. Except in the case of compact blocks, then we'd need to send messages for requesting the missing tx id's from peers. Even then, the pbft algorithm would only need to know if the tx exists and what id it is. Not the actual bytes or payload of it.

Reply:
This commit that goes into that direction: https://github.com/FuelLabs/fuel-pbft/pull/1/commits/0b5f3bf17a5a33d6342ade339873b0dbee6bee77